mlserver/env.py

```diff
@@ -0,0 +1,74 @@
+import asyncio
+import os
+import sys
+import tarfile
+import glob
+
+from typing import List
+from functools import cached_property
+
+from .logging import logger
+
+
+def _extract_env(tarball_path: str, env_path: str) -> None:
+    logger.info(f"Extracting environment tarball from {tarball_path}...")
+    with tarfile.open(tarball_path, "r") as tarball:
+        tarball.extractall(path=env_path)
+
+
+class Environment:
```

Comment:
question: How extensible is this class? for example if we wanted `Environment` to be defined from a `conda.yaml` or `requrirements.txt` or `name` etc.? It might be for a separate PR but we probably need to have a factory method, similar to `from_tarball` but more general...